tests/integration/test_authd/force_options/test_authd_force_options.py

```diff
@@ -0,0 +1,201 @@
+import os
+import time
+import pytest
+import yaml
+from wazuh_testing.tools import CLIENT_KEYS_PATH
+from wazuh_testing.tools.monitoring import make_callback, AUTHD_DETECTOR_PREFIX
+from wazuh_testing.tools.configuration import load_wazuh_configurations, set_section_wazuh_conf, write_wazuh_conf
+from wazuh_testing.tools.file import read_yaml
+from wazuh_testing.wazuh_db import query_wdb
+from authd import create_authd_request, validate_authd_response, AUTHD_KEY_REQUEST_TIMEOUT
+
+
+# Data paths
+data_path = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'data')
+configurations_path = os.path.join(data_path, 'configuration.yaml')
+tests_path = os.path.join(data_path, 'test_cases')
+
+# Configurations
+configurations = load_wazuh_configurations(configurations_path, __name__)
+configuration_ids = ['authd_force_options']
+
+# Tests
+tests = []
+test_case_ids = []
+for file in os.listdir(tests_path):
+    group_name = file.split(".")[0]
+    file_tests = read_yaml(os.path.join(tests_path, file))
+    tests = tests + file_tests
+    test_case_ids = test_case_ids + [f"{group_name} {test_case['name']}" for test_case in file_tests]
+
+# Variables
+log_monitor_paths = []
+
+receiver_sockets_params = [(("localhost", 1515), 'AF_INET', 'SSL_TLSv1_2')]
+monitored_sockets_params = [('wazuh-authd', None, True), ('wazuh-db', None, True)]
+receiver_sockets, monitored_sockets, log_monitors = None, None, None  # Set in the fixtures
+
+# Functions
+def validate_authd_logs(logs):
+    for log in logs:
+        log_monitor.start(timeout=AUTHD_KEY_REQUEST_TIMEOUT,
+                          callback=make_callback(log, prefix=AUTHD_DETECTOR_PREFIX,
+                                                 escape=True),
+                          error_message='Expected error log does not occured.')
```

Comment:
This could be useful not only for authd tests. Maybe it could be a good idea making it more general and move it to monitoring module.